app/controllers/answers_controller.rb

```diff
@@ -0,0 +1,53 @@
+class AnswersController < ApplicationController
+  before_action :authenticate_user!, except: [:show, :index]
+
+  def new
+    if current_user
+      @question = Question.find(params[:question_id])
+      @answer = @question.answers.new
+    else
+      flash[:error] = "You must be signed in"
+      redirect_to signin_path
+    end
+  end
+
+  def create
+    @question = Question.find(params[:question_id])
+    @answer = @question.answers.new(answer_params)
+
+    if @answer.save
+      redirect_to question_path(@question)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @question = Question.find(params[:question_id])
+    @answer = @question.answers.find(params[:id])
```

Comment:
If you load the answer you don't need to load the question - the association has it.